### PR TITLE
chore: remove fabric dependency

### PR DESCRIPTION
**Pull request by danielskinstad**
Follow up from QA-650 where we removed Fabric dependency in the integration repo. This allows mender-configure-module to be updated, since it doesn't have Fabric, and it uses this repo as a submodule.

Ticket: QA-666

### CODE REVIEW

File: helpers.py [hunk 1]
```diff
@@ -116,22 +164,11 @@ def __exit__(self, arg1, arg2, arg3):
...
 def new_tester_ssh_connection(setup_test_container):
-    config_hide = Config()
-    config_hide.run.hide = True
     with Connection(
         host="localhost",
         user=setup_test_container.user,
         port=setup_test_container.port,
-        config=config_hide,
-        connect_kwargs={
-            "key_filename": setup_test_container.key_filename,
```
Comment (lluiscampos) [inline helpers.py:127]: We need to preserve this, right? The key filename still needs to be specified somehow.

File: backend-integration-testing/requirements-py3.txt [hunk 2]
```diff
@@ -7,7 +7,6 @@ chardet==5.2.0
 cryptography==42.0.5
 docker==7.0.0
 execnet==2.1.1
-fabric==3.2.2
```
Comment (lluiscampos) [inline backend-integration-testing/requirements-py3.txt:10]: Good catch! This should be in a separate commit, though :monocle_face:

File: helpers.py [hunk 3]
```diff
@@ -21,9 +21,57 @@
...
+                raise ConnectionError(f"Could not connect using command '{command}'")
+
+            stdout = proc.stdout.decode()
+            stderr = proc.stderr.decode()
+
+            if not hide:
+                print(stdout)
+                print(stderr)
+
+            return Result(stdout, stderr, returncode)
```
Comment (lluiscampos) [inline helpers.py:71]: The code is in essence the same as in `integration` repository :+1: The main change is not accepting `connect_kwargs` for the filename but as I commented already I think we need to add that.

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Thank you for doing this.

Have you tested it with `mender-configure-module`? You can launch a PR there updating together integration git submodule and this one and see if CI likes it :robot:

File: helpers.py (hunk 1 above)
Comment (danielskinstad) [reply helpers.py:127]: I did run this with the tests in mender-configure-module locally, and since the the commands set StrictHostKeyChecking=no, I figured the key was unnecessary. But I can of course add it back if you think we need it.

Comment (danielskinstad) [reply helpers.py:127]: CI approves, https://gitlab.com/Northern.tech/Mender/mender-configure-module/-/pipelines/1271816938, so it doesn't seem like the keys are used

Comment (lluiscampos) [reply helpers.py:127]: Excellent. Good observation and great that we verified it locally and CI.

**Review by lluiscampos — APPROVED** [review]

File: backend-integration-testing/requirements-py3.txt (hunk 2 above)
Comment (lluiscampos) [reply backend-integration-testing/requirements-py3.txt:10]: Please split it into separate commits before merge.
